modules/ui.py

```diff
@@ -97,31 +100,88 @@ def save_files(js_data, images, index):
     filenames = []
 
     data = json.loads(js_data)
-    
+
     if index > -1 and opts.save_selected_only and (index > 0 or not opts.return_grid): # ensures we are looking at a specific non-grid picture, and we have save_selected_only
         images = [images[index]]
         data["seed"] += (index - 1 if opts.return_grid else index)
 
-    with open(os.path.join(opts.outdir_save, "log.csv"), "a", encoding="utf8", newline='') as file:
+    with open(
+        os.path.join(opts.outdir_save, "log.csv"),
+        "a",
+        encoding="utf8",
+        newline=''
+    ) as file:
         at_start = file.tell() == 0
         writer = csv.writer(file)
         if at_start:
-            writer.writerow(["prompt", "seed", "width", "height", "sampler", "cfgs", "steps", "filename", "negative_prompt"])
-
-        filename_base = str(int(time.time() * 1000))
+            writer.writerow([
+                "prompt",
+                "seed",
+                "width",
+                "height",
+                "sampler",
+                "cfgs",
+                "steps",
+                "filename",
+                "negative_prompt",
+            ])
+
+        # filename_base = str(int(time.time() * 1000))
         for i, filedata in enumerate(images):
-            filename = filename_base + ("" if len(images) == 1 else "-" + str(i + 1)) + ".png"
-            filepath = os.path.join(opts.outdir_save, filename)
-
-            if filedata.startswith("data:image/png;base64,"):
-                filedata = filedata[len("data:image/png;base64,"):]
-
-            with open(filepath, "wb") as imgfile:
-                imgfile.write(base64.decodebytes(filedata.encode('utf-8')))
-
-            filenames.append(filename)
-
-        writer.writerow([data["prompt"], data["seed"], data["width"], data["height"], data["sampler"], data["cfg_scale"], data["steps"], filenames[0], data["negative_prompt"]])
+            # needs_dash = ("" if len(images) == 1 else "-" + str(i + 1))
+            # filename = f"{filename_base}{needs_dash}.png"
+
+            image = image_from_url_text(filedata)
+            p = StableDiffusionProcessing(
+                sd_model=data.get('sd_model'),
+                prompt=data['prompt'],
+                seed=data['seed'],
+                width=data['width'],
+                height=data['height'],
+                sampler_index=data['sampler_index'],
+                steps=data['steps'],
+                cfg_scale=data['cfg_scale'],
+                restore_faces=data['restore_faces'],
+                tiling=data.get('tiling'),
+                negative_prompt=data['negative_prompt'],
+                styles=[],
+                seed_resize_from_w=0,
+                seed_resize_from_h=0,
+                denoising_strength=data.get('denoising_strength', 0.75),

```

Comment:
```suggestion
                subseed=(None if data['subseed_strength'] == 0 else data['subseed']),
                subseed_strength=(None if data['subseed_strength'] == 0 else data['subseed_strength']),
                denoising_strength=data.get('denoising_strength', 0.75),

```

Reply:
These suggestions will add in some missing data to regenerate the saved image, as shown in the UI.